src/libstd/sys/unix/backtrace/tracing/gcc_s.rs

```diff
@@ -85,25 +117,70 @@ pub fn write(w: &mut Write) -> io::Result<()> {
 
         // Don't print out the first few frames (they're not user frames)
         cx.idx += 1;
-        if cx.idx <= 0 { return uw::_URC_NO_REASON }
         // Don't print ginormous backtraces
         if cx.idx > 100 {
-            match write!(cx.writer, " ... <frames omitted>\n") {
+            match writeln!(cx.writer, " ... <frames omitted>") {
                 Ok(()) => {}
                 Err(e) => { cx.last_error = Some(e); }
             }
             return uw::_URC_FAILURE
         }
 
+        if cx.format == ::sys_common::backtrace::PrintFormat::Short {
+            if cx.skip_remaining_frames {
+                cx.skipped_end += 1;
+                return uw::_URC_NO_REASON;
+            }
+
+            if let Some(mangled_symbol_name) = resolve_symname(symaddr) {
+                let magics_begin = [
+                    "_ZN3std3sys3imp9backtrace7tracing3imp5write",
+                    "_ZN3std9panicking",
```

Comment:
This logic seems like we'd want to do this across all platforms, right? Perhaps this could go into the common module?